### ISSUE TITLE
"Save" Button has "Save and New" functionality in the Admin View Item

### ISSUE THREAD

**Issue by LordRazen** [vdm-io/Joomla-Component-Builder#686]
### Steps to reproduce the issue
- Create a component with an new Admin View

### Expected result
- If you create a new item in the Admin View, the "Save" button saves the data and return to the admin item view with the same data loaded

### Actual result
- The item is saved and you return to a new item. Basically the "Save and New" button functionality is on the "Save" button

### System information (as much as possible)
- OS Name & Version: Windows 10
- MySql Version: MariaDB 10.4.11
- Apache Version: Apache 2.4.43
- PHP Version: PHP 7.4.6
- Joomla Version: Joomla! 3.9.25 Stable
- JCB Version: 2.12.8
- Browser: Firefox

**Comment by Llewellynvdm** [comment]
Very interesting... those buttons are not managed by JCB directly... that is if I understand you correctly.

I do not get the same behavior... can you send me some screen shots? so I can visually see the issue step by step?

**Comment by LordRazen** [comment]
I tested the problem with the compiled component in a clean environment and could reproduce it. I've send the component to your github@ email adress for further analysis. Thanks a lot already ^^

**Comment by Llewellynvdm** [comment]
hmm I would prefer to see what you are trying to do, and make sure I understand the issue before anything else. So please lets start with the screenshots that show step by step what you are trying to do, or a video... if that is easier. But I need to first confirm that I understand you correctly.

**Comment by LordRazen** [comment]
Alright, here you go ^^
![SaveButton Problem](https://user-images.githubusercontent.com/9381368/110204946-1fd18e80-7e76-11eb-87bd-2ea2a9319d3e.gif)

**Comment by Llewellynvdm** [comment]
Okay this looks like a naming convention conflict, so I will need to look at the JCB, or Joomla package on this extension to see what is wrong. You mentioned that you emailed it to me, yet I did not receive anything.

When you compile the demo component that ships with JCB does it also have the same behavior?

Sine if it does then we have a systems issue, meaning your system may be the cause. Should the demo component not have this issue, the we know it is the component you build... and we should look there.

I am still not getting this kind of behavior in any of my projects, and I have many...

**Comment by LordRazen** [comment]
https://www.file-upload.net/download-14510928/com_mchentitymodels_v1_0_6__J3.zip.html
Here you go.

**Comment by Llewellynvdm** [comment]
When you compile the demo component that ships with JCB does it also have the same behavior?

**Comment by LordRazen** [comment]
o_O It doesnt. 

So seems like it's not necessarily a JCB error, but I still wondered since it happened with a version change. So I imported the JCB Package to my online test environment where there's still JCB 2.12.6. Compiled and the error also happened. 

Now I still wonder: Where would I have to search for the problem? I went through all the JCB settings and checked if everything is correct... it seemed like it...

**Comment by Llewellynvdm** [comment]
I think this is a naming convention clash... try renaming your view, and remove any custom JavaScript you may have placed on the page.

**Comment by LordRazen** [comment]
Did so. Renamed the views, dont have any custom JavaScript - no results. 

I tried to debug, but I'm new to that area of Joomla. I think the id of the edited item gets lost, but I dont know where. Any help is welcome here ^^

**Comment by LordRazen** [comment]
I did some further research and found that somehow, the System Name, Name and Name in Code are responsible or related to the bug. Sometimes I can get the name right and everything works (on a new dummy component), sometimes I get the bug. 

Are there any naming conventions for those names?

**Comment by LordRazen** [comment]
I found the bug: 

https://docs.joomla.org/J3.x:File_Structure_and_Naming_Conventions
`There are reserved words which can't be used in names of classes and components.
An example is the word "model" (in any case) for model classes (except "model" that must be second part of that class name)`

The "Name in Code" included the word "Model" within my component. 

I recommend to add this little detail to the tooltip. 

It really breaks your neck if you're not 100% familiar with the Joomla. I searched for this for like 5 days now...
